src/Components/Blazor/WebAssembly.Authentication/src/Models/RemoteAuthenticationStatus.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// Represents the status of an authentication operation.
+    /// </summary>
+    public class RemoteAuthenticationStatus
+    {
+        /// <summary>
+        /// The application is going to be redirected.
+        /// </summary>
+        public const string Redirect = "redirect";
+        /// <summary>
+        /// The authentication operation completed successfully.
+        /// </summary>
+        public const string Success = "success";
+        /// <summary>
+        /// There was an error performing the authentication operation.
+        /// </summary>
+        public const string Failure = "failure";
+
+        /// <summary>
+        /// The operation in the current navigation context has completed. This signals that the application running on the
+        /// current browser context is about to be shut down and no other work is required.
```

Comment:
Going to be shut down? What does that mean for a WebAssembly application?

Does it mean we know we're about to redirect away? If so it would be clearer to phrase it that way.

Reply:
I need more clarification. Are you expecting the Blazor application itself to be inside an iframe? Or if not, how could this be applicable?

Reply:
The blazor app can be inside an iframe during silent logins. I'll check if we need to do more here as part of https://github.com/dotnet/aspnetcore/issues/19053 if the host app is also within an iframe.

I have some ideas on what we can do here to simplify things.